graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -388,8 +541,170 @@ def enter_field_definition(
+    ) -> None:
+        """If the node's name matches the query type, record that we left the query type."""
+        self.current_type = None
+
+    def enter_field_definition(
+        self,
+        node: FieldDefinitionNode,
+        key: Any,
+        parent: Any,
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """If not at query type, check that no field depends on a type that was suppressed."""
+        if self.current_type == self.query_type:
+            return None
+        # At a field of a type that is not the query type
+        field_name = node.name.value
+        field_type = get_ast_with_non_null_and_list_stripped(node.type).name.value
+        if self.renamings.get(field_type, field_type):
+            return None
+        # Reaching this point means this field is of a type to be suppressed.
+        if self.current_type is None:
+            raise AssertionError(
+                "Entered a field not in any ObjectTypeDefinition scope because "
+                "self.current_type is None"
+            )
+        if self.current_type == field_type:
+            # Then node corresponds to a field belonging to type T that is also of type T.
+            # Therefore, we don't need to explicitly suppress the field as well and this should not
+            # raise errors.
+            return None
+        if self.current_type not in self.fields_to_suppress:
+            self.fields_to_suppress[self.current_type] = {}
+        self.fields_to_suppress[self.current_type][field_name] = field_type
+        return None
+
+    def enter_union_type_definition(
+        self,
+        node: UnionTypeDefinitionNode,
+        key: Any,
+        parent: Any,
+        path: List[Any],
+        ancestors: List[Any],
+    ) -> None:
+        """Check that each union still has at least one member."""
+        union_name = node.name.value
+        if node.types is None:
+            # This shouldn't happen, but the GraphQL core library type-hints the field as
+            # Optional[FrozenList[NamedTypeNode]]. Follow-up discussion here:
+            # https://github.com/graphql-python/graphql-core/issues/98
+            raise AssertionError(
+                f"Union {union_name} must include one or more unique member types, but "
+                f"{union_name}'s list of union types was None."
+            )
+        # Check if all the union members are suppressed.
+        for union_member in node.types:
+            union_member_type = get_ast_with_non_null_and_list_stripped(union_member).name.value
+            if self.renamings.get(union_member_type, union_member_type):
+                # Then at least one member of the union is not suppressed, so there is no cascading
+                # suppression error concern.
+                return None
+        if self.renamings.get(union_name):
+            # If the union is also suppressed, then nothing needs to happen here
+            return None
```

Comment:
Is this intentional? This conditional looks like it's checking for the exact opposite of what the comment is saying.

Also, a small nit: it's somewhat nicer to be explicit and say we're checking for `None` in case there's ever a falsy value in there, like the empty string.

Which reminds me, do you ever check that the renamed names are non-empty, satisfy our naming rules (i.e. no illegal chars), and do not duplicate each other or other existing types?

Reply:
fixed!

`check_type_name_is_valid` does check that the renamed names are valid and whether `renamings` duplicates anything, but checking against existing types only happens to a limited extent in `_rename_and_suppress_types`. I've noted all the improvements for error checking described here and plan to consolidate them all into `_validate_renamings` in a future PR just to keep this one at a manageable size.